### PR TITLE
pthread_barrier_wait: properly check the wait condition

### PR DESCRIPTION
**Pull request by Youw**
Fixes: #643

### CODE REVIEW

**Comment by Youw** [comment]
Yeap, verified with Coverity - that fixes the issue.

**Comment by Youw** [comment]
@z3ntu any chance you know why `builds.sr.ht` fails?

File: mac/hid.c [hunk 1]
```diff
@@ -83,16 +83,18 @@ static int pthread_barrier_wait(pthread_barrier_t *barrier)
 {
 	pthread_mutex_lock(&barrier->mutex);
 	++(barrier->count);
-	if(barrier->count >= barrier->trip_count)
-	{
+	if (barrier->count >= barrier->trip_count) {
 		barrier->count = 0;
 		pthread_cond_broadcast(&barrier->cond);
 		pthread_mutex_unlock(&barrier->mutex);
```
Comment (Youw) [inline mac/hid.c:89]:
```suggestion
		pthread_mutex_unlock(&barrier->mutex);
		pthread_cond_broadcast(&barrier->cond);
```

File: mac/hid.c [hunk 2]
```diff
@@ -83,16 +83,18 @@ static int pthread_barrier_wait(pthread_barrier_t *barrier)
...
 		return 1;
 	}
-	else
-	{
-		pthread_cond_wait(&barrier->cond, &(barrier->mutex));
+	else {
+		do {
+			pthread_cond_wait(&barrier->cond, &(barrier->mutex));
+		}
+		while (!(barrier->count >= barrier->trip_count));
```
Comment (Youw) [inline mac/hid.c:96]:
```suggestion
		while (barrier->count != 0);
```

**Comment by Youw** [comment]
still need to verify it locally though...

**Comment by Youw** [comment]
Try if now is enough

**Comment by z3ntu** [comment]
image

No, not even in this repository I have much access and nothing in the libusb organization (where the github app installations are). https://github.com/organizations/libusb/settings/installations . IIRC it's pretty high permissions needed for doing this, and on the organization level, not repository level..

Or we can do it like last time, request install of the app and then I do the sr.ht oauth setup, then that should be done, see https://github.com/libusb/hidapi/issues/462#issuecomment-1294134801 and following :) 

So: organization admin goes to the GitHub apps settings, then uninstalls the current install, goes to https://hottub.emersion.fr/ and reinstall. It should go to sr.ht for login, either give me this url or maybe even better, the url for the settings of the github app (which includes some numeric ID in the url). Then I reauthenticate from my side using that.

**Comment by Youw** [comment]
After uninstalling/installing the hottub bridge I immediatelly getting redirected to Authorisation page.
I've emailed you the authorisation link. See if that would work.

**Comment by Youw** [comment]
All good with `builds.sr.ht`.
Thanks @z3ntu!

**Comment by z3ntu** [comment]
Nice. Hopefully the refresh token things work by the time the token expires, if not we'll know what we have to do 1 year from now ;)

**Pull request merged**